app/main.py

```diff
@@ -1,34 +1,171 @@
+                    row_start,
+                    col_start,
+                    col_end
+                )
+            elif col_start == col_end:
+                if row_start > row_end:
+                    row_start, row_end = row_end, row_start
+                self._validate_vertical_ship_position(
+                    col_start,
+                    row_start,
+                    row_end
+                )
+            else:
+                raise ValueError
+
+    def check_h_position(self, row: int, index: int) -> None:
+        if not all(self.field[row + di][index] == self.EMPTY_CELL
+                   for di in [-1, 0, 1] if 0 <= row + di <= 9):
+            raise ValueError
+
+    def _validate_horizontal_ship_position(self,
+                                           row: int,
+                                           col: int
+                                           ) -> None:
+
+        if col == 0:
+            self.check_h_position(row, 0)
+            self.check_h_position(row, 1)
+
+        elif col == 9:
+            self.check_h_position(row, 9)
+            self.check_h_position(row, 8)
+        else:
+            if not all(self.field[row + di][col + dj] == self.EMPTY_CELL
+                       for di in [-1, 0, 1] for dj in [-1, 0, 1]
+                       if 0 <= row + di <= 9 and 0 <= col + dj <= 9):
+                raise ValueError
+
+    def check_v_position(self, col: int, index: int) -> None:
+        if not all(self.field[index][col + dj] == self.EMPTY_CELL
+                   for dj in [-1, 0, 1] if 0 <= col + dj <= 9):
+            raise ValueError
+
+    def _validate_vertical_ship_position(self,
+                                         col: int,
+                                         row: int,
+                                         ) -> None:
+
+        if row == 0:
+            self.check_v_position(col, 0)
+            self.check_v_position(col, 1)
+        elif row == 9:
+            self.check_v_position(col, 9)
+            self.check_v_position(col, 8)
+        else:
+            if not all(self.field[row + di][col + dj] == self.EMPTY_CELL
+                       for di in [-1, 0, 1] for dj in [-1, 0, 1] if
+                       0 <= row + di <= 9 and 0 <= col + dj <= 9):
+                raise ValueError
+
+    def print_field(self) -> None:
+        for row in self.field:
+            for item in row:
+                print(item, end=" ")
+            print()
+
+    def _add_ship(self, start: tuple[int, int], end: tuple[int, int]) -> None:
+        ship = Ship(start, end)
+        self.ships.append(ship)
+        for deck in ship.decks:
+            self.field[deck.row][deck.column] = self.SHIP_DECK
+
+    def _get_ship(self, row: int, column: int) -> Ship:
+        for ship in self.ships:
+            if (len(ship.decks) == 1 and row == ship.start[0]
+                    and column == ship.start[1]):
+                return ship
```

Comment:
Maybe you can check if coord in ship because for now check is so hard
Try to simplify it